pkg/cmd/util/clientcmd/factory.go

```diff
@@ -192,9 +192,17 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 				return nil, errors.New("provided options object is not a BuildLogOptions")
 			}
 			if bopts.Version != nil {
-				// should --version work with builds at all?
 				return nil, errors.New("cannot specify a version and a build")
 			}
+			if bopts.Previous {
```

Comment:
If we are going to support `oc logs build/golang-2 --previous`, this should probably happen in the apiserver, right @csrwng @bparees ?